erdpy/delegation/staking_provider.py

```diff
@@ -0,0 +1,141 @@
+import binascii
+from typing import Any
+from os import path
+
+from erdpy.accounts import Account
+from erdpy.config import MetaChainSystemSCsCost
+from erdpy.utils import read_json_file_validators, parse_keys
+from erdpy.validators.core import estimate_system_sc_call
+from erdpy.wallet.pem import parse_validator_pem
+from erdpy.wallet.signing import sign_message_with_bls_key
+
+DELEGATION_MANAGER_SC_ADDRESS = "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6"
+
+
+def prepare_args_for_create_new_staking_contract(args: Any):
+    args.data = 'createNewDelegationContract'
+    args.data += '@' + string_to_hex_bytes(str(args.total_delegation_cap))
+    args.data += '@' + string_to_hex_bytes(str(args.service_fee))
+
+    args.receiver = DELEGATION_MANAGER_SC_ADDRESS
+
+    if args.estimate_gas:
+        args.gas_limit = estimate_system_sc_call(args, MetaChainSystemSCsCost.DELEGATION_MGR_OPS, 3)
+
+
+def string_to_hex_bytes(number_str: str) -> str:
+    num_of_bytes = 1
+    if len(number_str) > 2:
+        num_of_bytes = int(len(number_str) / 2)
+    int_str = int(number_str)
+    int_bytes = int_str.to_bytes(num_of_bytes, byteorder="big")
+    bytes_str = binascii.hexlify(int_bytes).decode()
+    return bytes_str
+
+
+def prepare_args_for_add_nodes(args: Any):
+    validators_file = args.validators_file
+    validators_data = read_json_file_validators(validators_file)
+
+    # TODO: Refactor, so that only address is received here.
+    if args.pem:
+        account = Account(pem_file=args.pem)
+    elif args.keyfile and args.passfile:
+        account = Account(key_file=args.keyfile, pass_file=args.passfile)
+
+    add_nodes_data = "addNodes"
+    num_of_nodes = len(validators_data.get("validators", []))
+    for validator in validators_data.get("validators", []):
+        # get validator
+        validator_pem = validator.get("pemFile")
+        validator_pem = path.join(path.dirname(validators_file), validator_pem)
+        seed, bls_key = parse_validator_pem(validator_pem)
```

Comment:
this will return the same validator key. Is this the intended behavior?

Reply:
We should at least add a description stating that we support only one key in the pem file and it's ok

Reply:
this will read every validator pem that is specified in `validators_file.json`

Reply:
correct, but if those pems contain multiple keys, only the first ones will be read